File: labs/py/01.py
def convertYears(x):
    k1 = 10.5
    k2 = 4
    if x < 0:    raise ValueError("Years should be positive!")
    elif x <= 2: return x * k1
    else:        return 2 * k1 + (x - 2) * k2

humanYears = float(input("Enter human years: "))
try:
    dogYears = convertYears(humanYears)
    print("Dog years:", dogYears)
except ValueError as e:
    print(e)

File: labs/hs/01.hs
convertYears :: Double -> Either String Double
convertYears x
    | x < 0     = Left "Years should be positive!"
    | x <= 2    = Right $ x * k1
    | otherwise = Right $ 2 * k1 + (x - 2) * k2
    where { k1 = 10.5; k2 = 4 }

main :: IO ()
main = do
    putStrLn "Enter human years: "
    y <- convertYears <$> readLn
    putStrLn $ either id (("Dog years: " ++) . show) y